File: backend/app/schemas/portfolio_valuation.py
from datetime import datetime
from decimal import Decimal
from typing import Optional
from uuid import UUID
from pydantic import BaseModel


class PortfolioValuationResponse(BaseModel):
    execution_mode: str
    user_id: UUID
    broker_id: Optional[UUID] = None
    paper_portfolio_id: Optional[UUID] = None
    cash_balance: Optional[Decimal] = None
    market_value: Decimal
    realized_pnl: Decimal
    unrealized_pnl: Decimal
    total_pnl: Decimal
    equity: Decimal
    position_count: int
    valued_at: datetime

File: frontend/src/types/portfolioValuation.ts
export interface PortfolioValuation {
  execution_mode: 'LIVE' | 'PAPER';
  user_id: string;
  broker_id?: string | null;
  paper_portfolio_id?: string | null;
  cash_balance?: string | null;
  market_value: string;
  realized_pnl: string;
  unrealized_pnl: string;
  total_pnl: string;
  equity: string;
  position_count: number;
  valued_at: string;
}
